cardano-crypto-tests/bls12-381-test-vectors/test-vectors-bls12_381/src/main.rs

```diff
@@ -0,0 +1,338 @@
+    let hex_G1_ADD = hex::encode(G1_ADD.to_compressed());
+    let hex_G1_SUB = hex::encode(G1_SUB.to_compressed());
+    let hex_G1_MUL = hex::encode(G1_MUL.to_compressed());
+    let hex_G1_NEG = hex::encode(G1_NEG.to_compressed());
+
+    let hex_G2_P = hex::encode(G2_P.to_compressed());
+    let hex_G2_Q = hex::encode(G2_Q.to_compressed());
+    let hex_G2_ADD = hex::encode(G2_ADD.to_compressed());
+    let hex_G2_SUB = hex::encode(G2_SUB.to_compressed());
+    let hex_G2_MUL = hex::encode(G2_MUL.to_compressed());
+    let hex_G2_NEG = hex::encode(G2_NEG.to_compressed());
+
+    let mut file = File::create("ec_operations_test_vectors")?;
+    file.write_all(hex_G1_P.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G1_Q.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G1_ADD.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G1_SUB.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G1_MUL.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G1_NEG.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G2_P.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G2_Q.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G2_ADD.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G2_SUB.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G2_MUL.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_G2_NEG.as_ref())?;
+
+    Ok(())
+}
+
+fn serde<R: RngCore>(mut rng: R) -> std::io::Result<()> {
+    //---- G1----
+    let G1_P = G1Affine::from(G1Projective::random(&mut rng));
+    let mut G1_bytes = G1_P.to_uncompressed();
+    assert_eq!(
+        G1Affine::from_uncompressed(&G1_bytes).is_some().unwrap_u8(),
+        1
+    );
+    G1_bytes[4] ^= 1;
+    assert_eq!(
+        G1Affine::from_uncompressed(&G1_bytes).is_none().unwrap_u8(),
+        1
+    );
+
+    let G1_uncomp_curve_hex = hex::encode(G1_bytes);
+
+    let mut G1_compressed = G1_P.to_compressed();
+    G1_compressed[4] ^= 1;
+    assert_eq!(
+        G1Affine::from_compressed(&G1_compressed)
+            .is_none()
+            .unwrap_u8(),
+        1
+    );
+
+    let G1_comp_curve_hex = hex::encode(G1_compressed);
+
+    let mut G1_random_bytes = [0u8; 48];
+    for _ in 0..10 {
+        rng.fill_bytes(&mut G1_random_bytes);
+        G1_random_bytes[0] |= 0b10000000;
+        G1_random_bytes[0] &= 0b10011111;
+        let G1_try_out_group = G1Affine::from_compressed_unchecked(&G1_random_bytes);
+        if G1_try_out_group.is_some().unwrap_u8() == 1
+            && G1_try_out_group.unwrap().is_torsion_free().unwrap_u8() == 0
+        {
+            assert_eq!(
+                G1Affine::from_compressed(&G1_random_bytes)
+                    .is_none()
+                    .unwrap_u8(),
+                1
+            );
+            break;
+        }
+    }
+
+    let G1_comp_group_hex = hex::encode(G1_random_bytes);
```

Comment:
This should happen inside the loop, otherwise, it could happen that no point is found and we store a slice of zeroes.